<h1  align="center"><b> ANÁLISE FINAL </b></h1>

`Objetivo Geral:` Importar os dados pluviométricos do município de São Paulo e as previsões para os 7 primeiros dias de 2021 para determinar qual o melhor modelo dentre os testados.

`Dados:` Os dados foram retirados do [Banco de Dados Hidrológicos](http://www.hidrologia.daee.sp.gov.br/) do Portal do Departamento de Águas e Energia Elétrica do Estado de São Paulo (DAEE) no dia 21/12/2023 e as previsões foram retiradas deste repositório.

In [17]:
import pandas as pd
from sklearn.metrics import mean_absolute_error, mean_squared_error

<h2 align="center"><b> Criando a Tabela </b></h2>

In [20]:
previsão_AR = pd.read_csv('../[4] Modelo AR - Projeto Chuva/Previsão AR - Chuva Mensal.csv', sep = ';', index_col = 0, skipfooter = 5, engine = 'python')

previsão_MA = pd.read_csv('../[5] Modelo MA - Projeto Chuva/Previsão MA - Chuva Mensal.csv', sep = ';', index_col = 0, skipfooter = 5, engine = 'python')

previsão_ARMA = pd.read_csv('../[6] Modelo ARMA - Projeto Chuva/Previsão ARMA - Chuva Mensal.csv', sep = ';', index_col = 0, skipfooter = 5, engine = 'python')

previsão_ARIMA =  pd.read_csv('../[7] Modelo ARIMA - Projeto Chuva/Previsão ARIMA - Chuva Mensal.csv', sep = ';', index_col = 0, skipfooter = 5, engine = 'python')

previsão_AutoARIMA = pd.read_csv('../[8] Modelo AutoARIMA - Projeto Chuva/Previsão AutoARIMA - Chuva Mensal.csv', sep = ';', index_col = 0, skipfooter = 5, engine = 'python')

chuva_mensal = pd.DataFrame([377.6, 452.0, 138.2, 165.7, 43.3, 85.2, 15.0], columns = ['Valores Reais'], index = previsão_AR.index)

chuva_mensal = pd.concat([chuva_mensal, previsão_AR, previsão_MA, previsão_ARMA, previsão_ARIMA, previsão_AutoARIMA], axis = 1)

<h2 align="center"><b> Incluindo as Métricas </b></h2>

In [21]:
mae = {}
rmse = {}

for coluna in chuva_mensal.columns:
    if coluna != 'Valores Reais':
        mae[coluna] = mean_absolute_error(chuva_mensal['Valores Reais'], chuva_mensal[coluna])
        rmse[coluna] = mean_squared_error(chuva_mensal['Valores Reais'], chuva_mensal[coluna], squared = False)

mae = pd.DataFrame(mae, index = ['MAE'])
rmse = pd.DataFrame(rmse, index = ['RMSE'])
chuva_mensal = pd.concat([chuva_mensal, mae, rmse])

In [22]:
chuva_mensal = chuva_mensal.round(2) # Deixando apenas duas casas decimais
chuva_mensal.to_csv('Análise Final - Projeto Chuva.csv', header = True) # Exportando para csv
display(chuva_mensal) # Visualizando tabela

Valores Reais  Previsão AR(7)  Previsão MA(7)  \
2023-01-31          377.6          244.75          188.90   
2023-02-28          452.0          212.57          173.03   
2023-03-31          138.2          134.93          115.22   
2023-04-30          165.7           84.14          100.40   
2023-05-31           43.3           63.24          106.71   
2023-06-30           85.2           49.20           98.27   
2023-07-31           15.0           49.77          100.78   
MAE                   NaN           78.26          102.60   
RMSE                  NaN          109.90          136.16   

            Previsão ARMA(2, 2)  Previsão ARIMA(2, 1, 2)  Previsão AutoARIMA  
2023-01-31               270.32                   155.45              250.03  
2023-02-28               238.35                   127.95              212.70  
2023-03-31               171.12                   121.96              150.62  
2023-04-30               103.47                   112.37               92.76  
2023-05-31                57.45                   114.00               54.80  
2023-06-30                34.91                   108.51               36.93  
2023-07-31                29.28                   111.94               34.25  
MAE                       70.69                   115.25               75.89  
RMSE                      96.39                   156.94              108.13

<h2 align="center"><b> Conclusão </b></h2>

**1° Lugar: Modelo ARMA(2, 2)**
<br>&emsp;&emsp; **MAE:** 80.03
<br>&emsp;&emsp; **RMSE:** 96.39

**2° Lugar: Modelo  AutoARIMA**
<br>&emsp;&emsp; **MAE:** 87.61
<br>&emsp;&emsp; **RMSE:** 108.13
<br>&emsp;&emsp; **Obs:** O modelo AutoARIMA foi executado com o parâmetro stepwise = True, o que resultou em um tempo de execução significativamente menor durante essa etapa. No entanto, para que isso fosse possível, o algoritmo realiza uma busca menos minuciosa, o que fez com que ele ignorasse o modelo ARMA(2, 2) e/ou um possível modelo superior.

**3° Lugar: Modelo AR(7)**
<br>&emsp;&emsp; **MAE:** 89.77
<br>&emsp;&emsp; **RMSE:** 109.90

**4° Lugar: Modelo MA(7)**
<br>&emsp;&emsp; **MAE:** 114.80
<br>&emsp;&emsp; **RMSE:** 136.16

**5° Lugar: Modelo ARIMA(2, 1, 2)`**
<br>&emsp;&emsp; **MAE:** 130.41
<br>&emsp;&emsp; **RMSE:** 156.94
